sonar-batch/src/main/java/org/sonar/batch/mediumtest/TaskResult.java

```diff
@@ -197,29 +195,21 @@ public InputDir inputDir(String relativePath) {
    */
   public List<TypeOfText> highlightingTypeFor(InputFile file, int line, int lineOffset) {
     int ref = reportComponents.get(((DefaultInputFile) file).key()).getRef();
-    File highlightingFile = reader.readComponentSyntaxHighlighting(ref);
-    if (highlightingFile == null) {
+    if (!reader.hasSyntaxHighlighting(ref)) {
       return Collections.emptyList();
     }
     TextPointer pointer = file.newPointer(line, lineOffset);
     List<TypeOfText> result = new ArrayList<>();
-    InputStream inputStream = null;
-    try {
-      inputStream = FileUtils.openInputStream(highlightingFile);
-      BatchReport.SyntaxHighlighting rule = BatchReport.SyntaxHighlighting.PARSER.parseDelimitedFrom(inputStream);
-      while (rule != null) {
+    try (CloseableIterator<BatchReport.SyntaxHighlighting> it = reader.readComponentSyntaxHighlighting(ref)) {
+      while (it.hasNext()) {
```

Comment:
![BLOCKER](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-blocker.png) NullPointerException might be thrown as 'it' is nullable here [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS2259)